nix-shell fails to build apr-util

It looks like #4829 broke the `nix-shell` build (at least on MacOS):

```
error: Cannot build '/nix/store/g2rd4fxwsziaawbk2zg98lcaxlsj4qb6-apr-util-1.6.3.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/cq7iffsnasf1szvkla7d7p1vmyawjgr6-apr-util-1.6.3-dev
         /nix/store/k2b0y64zkw0qxnlf291wmabganm4lafy-apr-util-1.6.3
       Last 25 log lines:
       > dbm/sdbm/sdbm_pair.c:132:14: error: unknown type name 'key'
       >   132 | duppair(pag, key)
       >       |              ^
       > dbm/sdbm/sdbm_pair.c:132:18: error: expected ';' after top level declarator
       >   132 | duppair(pag, key)
       >       |                  ^
       >       |                  ;
       > dbm/sdbm/sdbm_pair.c:135:1: error: expected identifier or '('
       >   135 | {
       >       | ^
       > dbm/sdbm/sdbm_pair.c:141:9: error: unknown type name 'pag'
       >   141 | getnkey(pag, num)
       >       |         ^
       > dbm/sdbm/sdbm_pair.c:141:14: error: unknown type name 'num'; did you mean 'enum'?
       >   141 | getnkey(pag, num)
       >       |              ^~~
       >       |              enum
       > fatal error: too many errors emitted, stopping now [-ferror-limit=]
       > 20 errors generated.
       > /nix/store/f700nj7wlwg441h39gkq29qbviy99sgq-bash-5.3p9/bin/bash /nix/store/z9asvf3gxg3iqa3h7znnq42mmarhkymz-apr-1.7.6-dev/share/build/libtool --silent --mode=compile --tag=CC clang -std=gnu23 -g -O2   -DHAVE_CONFIG_H  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK   -I/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/include -I/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/include/private  -I/nix/store/z9asvf3gxg3iqa3h7znnq42mmarhkymz-apr-1.7.6-dev/include  -I/nix/store/w2zd8g8k3nqp1lg5niwdxnj06p9sdbi9-openssl-3.6.1-dev/include -I/nix/store/i6szhry1wgnijqsv9w88y0c6jh5zspji-db-5.3.28-dev/include -I/nix/store/ii1amx0m1jiy5qa0njn6ibd2av9blg0m-expat-2.7.5-dev/include  -o misc/apr_queue.lo -c misc/apr_queue.c && touch misc/apr_queue.lo
       > /nix/store/f700nj7wlwg441h39gkq29qbviy99sgq-bash-5.3p9/bin/bash /nix/store/z9asvf3gxg3iqa3h7znnq42mmarhkymz-apr-1.7.6-dev/share/build/libtool --silent --mode=compile --tag=CC clang -std=gnu23 -g -O2   -DHAVE_CONFIG_H  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK   -I/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/include -I/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/include/private  -I/nix/store/z9asvf3gxg3iqa3h7znnq42mmarhkymz-apr-1.7.6-dev/include  -I/nix/store/w2zd8g8k3nqp1lg5niwdxnj06p9sdbi9-openssl-3.6.1-dev/include -I/nix/store/i6szhry1wgnijqsv9w88y0c6jh5zspji-db-5.3.28-dev/include -I/nix/store/ii1amx0m1jiy5qa0njn6ibd2av9blg0m-expat-2.7.5-dev/include  -o misc/apr_reslist.lo -c misc/apr_reslist.c && touch misc/apr_reslist.lo
       > make[1]: *** [/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/build/rules.mk:207: dbm/sdbm/sdbm_pair.lo] Error 1
       > make[1]: *** Waiting for unfinished jobs....
       > make[1]: Leaving directory '/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3'
       > make: *** [/nix/var/nix/builds/nix-51166-1631853614/apr-util-1.6.3/build/rules.mk:119: all-recursive] Error 1
       For full logs, run:
         nix log /nix/store/g2rd4fxwsziaawbk2zg98lcaxlsj4qb6-apr-util-1.6.3.drv
error: Cannot build '/nix/store/20323lc0zrcsx13xb2y3nds887v9qyaz-subversion-1.14.5.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/6wsxn42659br0w9ljw36i2cmp1gwp5wd-subversion-1.14.5-dev
         /nix/store/d8ycspxb8y4p49j7inrir779nf1b9g9z-subversion-1.14.5-man
         /nix/store/fw65akab0kvygwkma54xw020pgmvfrlh-subversion-1.14.5
error: Cannot build '/nix/store/kf01yz6mw5waaglrj5w76isbhxq5rcfj-nix-prefetch-svn-26.05.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/ijxz7dhwy4rbccgv53q0as2f4bcxwyn1-nix-prefetch-svn-26.05
error: Cannot build '/nix/store/ba3p0vig0jk7m3l692v6zs56dfmbjl86-nix-prefetch-scripts.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/5gbl9hql0hi1wwr2iicxymd73mcpjccj-nix-prefetch-scripts
error: Cannot build '/nix/store/gi062hyz6ga8pbiv541haqks9k3lmfnm-cabal2nix-2.21.2.drv'.
       Reason: 1 dependency failed.
       Output paths:
         /nix/store/i5a6n7xqagjk47mwh7vxnhvj0yyiamwl-cabal2nix-2.21.2
error: build of '/nix/store/07p00ji9b0cw8xj5gl5jsw2rd46nsw8n-postgrest-with.drv', '/nix/store/0rc5qf0rab2cb9r8dyxmrci0hslp50a4-postgrest-loadtest.drv', '/nix/store/148c52zbmihm7hdngxhza7lwar6al759-postgrest-style.drv', '/nix/store/48qscx43kpg8vpxw5msrjf83jkmchg5v-hsie.drv', '/nix/store/gi062hyz6ga8pbiv541haqks9k3lmfnm-cabal2nix-2.21.2.drv', '/nix/store/nrh2fhnr4wv9rd8rgwj1lhlhb56gjw2h-postgrest-tests.drv', '/nix/store/r081d6s51ip1312rijca9cd017qfyy1k-hsie.drv', '/nix/store/swa2s41dir89h9bxm8khadb2zhi4ph4a-ghc-9.12.3-with-packages.drv', '/nix/store/z7d99s3m37il0nghwlrqfzw0j2c533f2-postgrest-cabal.drv', '/nix/store/z7ql9qh50ydv5fm7ii84qfnmwsrzn7z8-postgrest-dev.drv' failed

```

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -78,19 +78,11 @@ jobs:
         uses: ./.github/actions/setup-nix
         with:
           authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
-      - name: Install gnu sed
-        run: brew install gnu-sed
+      - name: Install nix-build-uncached
+        run: nix-env -f default.nix -iA nix-build-uncached
 
-      - name: Build everything
-        run: |
-          # The --dry-run will give us a list of derivations to download from cachix and
-          # derivations to build. We only take those that would have to be built and then build
-          # those explicitly. This has the advantage that pure verification will not include
-          # a download anymore, making it much faster. If something needs to be built, only
-          # the dependencies required to do so will be downloaded, but not everything.
-          nix-build --dry-run 2>&1 \
-            | gsed -e '1,/derivations will be built:$/d' -e '/paths will be fetched/Q' \
-            | xargs nix-build
+      - name: Build everything (default.nix)
+        run: nix-build-uncached
 
 
   stack:

--- a/.github/workflows/build.yaml
+++ b/.github/workflows/build.yaml
@@ -84,6 +84,9 @@ jobs:
       - name: Build everything (default.nix)
         run: nix-build-uncached
 
+      - name: Build everything (shell.nix)
+        run: nix-build-uncached shell.nix
+
 
   stack:
     strategy:

--- a/default.nix
+++ b/default.nix
@@ -107,6 +107,9 @@ rec {
       inherit (pkgs.haskell.packages."${compiler}") ghcWithPackages;
     };
 
+  # Used by CI on MacOS
+  inherit (pkgs) nix-build-uncached;
+
   ### Tools
 
   cabalTools =

--- a/shell.nix
+++ b/shell.nix
@@ -35,10 +35,7 @@ lib.overrideDerivation postgrest.env (
     buildInputs =
       base.buildInputs ++ [
         pkgs.cabal-install
-        pkgs.cabal2nix
-        pkgs.git
         pkgs.postgresql
-        pkgs.update-nix-fetchgit
         postgrest.hsie.bin
       ]
       ++ toolboxes;

